kernel_hardening_checker/engine.py

```diff
@@ -81,6 +81,22 @@ def set_state(self, data: StrOrNone) -> None:
         self.state = data
 
     def check(self) -> None:
+        # handle the *special* check
+        if self.expected.startswith('*') and self.expected.endswith('*'):
+            if self.state is not None:
+                if self.state.startswith('"') and self.state.endswith('"'):
+                    option = self.expected.strip('*')
+                    optlist  = list(self.state.strip('"').split(','))
+                    if option in optlist:
+                        self.result = f'OK: {self.state}'
+                    else:
+                        self.result = f'FAIL: {self.state}'
+                else:
+                    self.result = f'FAIL: {self.state}'
+            else:
+                self.result = f'FAIL: {self.state}'
```

Comment:
There is a bug here.
Please don't print `None` if the option is not found.
By the way, testing for this code is missing in the unit test.